rails/benchmarks/app.rb

```diff
@@ -1,290 +1,17 @@
-  end
-
-  # PATCH/PUT /comments/1
-  # PATCH/PUT /comments/1.json
-  def update
-    respond_to do |format|
-      if @comment.update(comment_params)
-        format.html { redirect_to @post, notice: 'Comment was successfully updated.' }
-        format.json { render :show, status: :ok, location: @post }
-      else
-        format.html { render :edit }
-        format.json { render json: @comment.errors, status: :unprocessable_entity }
-      end
-    end
-  end
-
-  # DELETE /comments/1
-  # DELETE /comments/1.json
-  def destroy
-    @comment.destroy
-    respond_to do |format|
-      format.html { redirect_to @post, notice: 'Comment was successfully destroyed.' }
-      format.json { head :no_content }
-    end
-  end
-
-  private
-
-  # Use callbacks to share common setup or constraints between actions.
-  def set_comment
-    @comment = @post.comments.find(params[:id])
-  end
-
-  def set_post
-    @post = Post.find(params[:post_id])
-  end
-
-  # Never trust parameters from the scary internet, only allow the white list through.
-  def comment_params
-    if defined?(ActionController::Parameters) && params.instance_of?(ActionController::Parameters)
-      params.require(:comment).permit(:body, :author, :email)
-    else
-      params[:comment]
-    end
-  end
-end
-
-class RouteNotFoundError < StandardError;end
-
-def request(method, path, query_string: "", body: {})
-  env = {
-    "GATEWAY_INTERFACE"=>"CGI/1.1",
-    "PATH_INFO"=>path,
-    "QUERY_STRING"=>query_string,
-    "REMOTE_ADDR"=>"127.0.0.1",
-    "REMOTE_HOST"=>"127.0.0.1",
-    "REQUEST_METHOD"=>method.to_s.upcase,
-    "REQUEST_URI"=>"http://localhost:3000#{path}",
-    "SCRIPT_NAME"=>"",
-    "SERVER_NAME"=>"localhost",
-    "SERVER_PORT"=>"3000",
-    "SERVER_PROTOCOL"=>"HTTP/1.1",
-    "SERVER_SOFTWARE"=>"WEBrick/1.3.1 (Ruby/2.1.2/2014-05-08)",
-    "HTTP_HOST"=>"localhost:3000",
-    "HTTP_ACCEPT_ENCODING"=>"gzip, deflate",
-    "HTTP_ACCEPT"=>"*/*",
-    "HTTP_USER_AGENT"=>"HTTPie/0.8.0",
-    "rack.version"=>[1, 2],
-    "rack.multithread"=>false,
-    "rack.multiprocess"=>false,
-    "rack.run_once"=>false,
-    "rack.url_scheme"=>"http",
-    "rack.input" => StringIO.new(body.to_query),
-    "HTTP_VERSION"=>"HTTP/1.1",
-    "REQUEST_PATH"=>path
-  }
-  response = BenchmarkApp.call(env)
-  if response[0].in?([404, 500])
-    raise RouteNotFoundError.new, "not found #{method.to_s.upcase} #{path}"
-  end
-  response
-end
-
-
-m = Benchmark.measure do
-  request(:get, "/posts")
+m = Benchmark.rails(100, "app") do
+  AppBenchmark.request(:get, "/posts")
   2.times do
-    request(:get, "/posts/new")
-    request(:post, "/posts", body: { post: { title: Faker::Food.herb_or_spice, body: Faker::HipsterIpsum.words(50).join(" "), author: Faker::Name.name }})
+    AppBenchmark.request(:get, "/posts/new")
+    AppBenchmark.request(:post, "/posts", body: { post: { title: Faker::Food.herb_or_spice, body: Faker::HipsterIpsum.words(50).join(" "), author: Faker::Name.name }})
```

Comment:
Just noticed this. We're going to be measuring the performance of `Faker` here though it might be insignificant.